PolyEngine/Core/Src/Collections/BitMask.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Dynarray.hpp"
+
+namespace Poly {
+	class CORE_DLLEXPORT BitMask
+	{
+	public:
+		using DataType = unsigned short;
+
+		BitMask(size_t size = sizeof(DataType));
+
+		//Bitwise operators
+		BitMask operator|(const BitMask& rhs) const;
+		BitMask operator^(const BitMask& rhs) const;
+		BitMask operator&(const BitMask& rhs) const;
+		BitMask& operator~();
+		//Bitwise assignment operators
+		BitMask& operator|=(const BitMask& rhs);
+		BitMask& operator^=(const BitMask& rhs);
+		BitMask& operator&=(const BitMask& rhs);
+
+
+		bool operator==(const BitMask rhs) const;
+		bool operator!=(const BitMask rhs) const
+		{
+			return !(*this == rhs);
+		}
```

Comment:
Put this whole method in one line. This should accept `const BitMask&` as argument